### ISSUE TITLE
Working Great!

### ISSUE THREAD

**Issue by jdtsmith** [russhughes/lv_st7789#1]
Not really an issue, just reporting back: This is excellent!  Rock solid in my testing thus far.  And of course LVGL is much better and less fragile than the kludgy multi-line text stuff I had before. 

I noticed no difference in speed or scrolling quality etc. with double-buffering or buffers factors below 4 (i.e. 1/4 screen buffer).  Not sure if SPI would use DMA with your driver, to make double-buffering worth it.  I'll compare for other operations.   

One bummer is the recent f-string addition to MP, which is in your other firmware, isn't pulled into LVGL yet, but should be soon.   It's also strange that it can't load JPG images directly, but requires pre-processing into a binary format.  It's a bit challenging to navigate the C docs and guess correspondence in MP, but getting the hang of it.  I even converted and loaded a large TTF font, like so:

```
lv_font_conv --font ~/Downloads/pacifico/Pacifico.ttf --bpp 1 --format bin --no-compress --size 95 -o fonts/pacifico_large.bin --symbols 'fedthdog '
```

And here's a scrolling large label with color coding:

```python
import lvgl as lv
import fs_driver

lv.init()

from ili9XXX import st7789
disp = st7789(width=135, height=240, double_buffer = False, factor=4, rot=st7789.LANDSCAPE)

scr = lv.scr_act()

# FS driver init.
fs_drv = lv.fs_drv_t()
fs_driver.fs_register(fs_drv, 'S')

# Load font
font=lv.font_load('S:fonts/pacifico_large.bin')
dark = lv.style_t()
dark.set_bg_color(lv.color_hex(0x747474))
scr.add_style(dark,lv.PART.MAIN)

l1 = lv.label(scr)
l1.set_style_text_font(font,0)
l1.set_recolor(True)
l1.set_text('#FF0000 feed# #00FF00 the# #0000FF dog#')
l1.set_long_mode(lv.label.LONG.SCROLL_CIRCULAR) # Circular scroll
l1.set_width(disp.width)
```

Thanks for your speedy efforts!  Hope you are enjoying LVGL on ST7789. This should definitely be submitted for inclusion over [here](https://github.com/lvgl/lv_binding_micropython).

**Comment by russhughes** [comment]
Good to hear it's working for you.  The driver should be using DMA as long as you have not set "hybrid=False".

**Comment by jdtsmith** [comment]
Have you found any speedup from double-buffering, i.e. from the DMA-while-drawing-to-the-other-buffer?  I tried things like setting the OPA(city) of a screen's bg image 255 times in succession, and found at most a couple percent speedup with double-buffering enabled (maybe not even significant).  I also tried `mhz=80` but that leads to artifacts; not sure if the SPI bus should be able to support those speeds in principle.  

BTW, things like calendar works great:

```python
import time
scr.clean()
t=time.localtime()
cal=lv.calendar(scr)
cal.set_size(disp.width,disp.height)
cal.align(lv.ALIGN.CENTER,0,0)
cal.set_today_date(*t[0:3])
cal.set_showed_date(t[0],t[1])
```

I wish I had some SPI RAM though; this eats it up.  I've dropped to `factor=8` single-buffering without apparent loss in speed.
